kf/nfconfig.go

```diff
@@ -942,3 +943,297 @@ func getHostInterfaces() (map[string]bool, error) {
 	}
 	return hostIfaces, nil
 }
+
+func (c *NFConfigs) AddAndStartBPF(bpfProg *models.BPFProgram, ifaceName string, direction string) error {
+	var bpfList *list.List
+	if bpfProg == nil {
+		return fmt.Errorf("AddAndStartBPF - bpf program is nil")
+	}
+
+	if bpfProg.AdminStatus == models.Disabled {
+		return nil
+	}
+
+	bpf := NewBpfProgram(c.ctx, *bpfProg, c.HostConfig)
+
+	switch direction {
+	case models.XDPIngressType:
+		bpfList = c.IngressXDPBpfs[ifaceName]
+	case models.IngressType:
+		bpfList = c.IngressTCBpfs[ifaceName]
+	case models.EgressType:
+		bpfList = c.EgressTCBpfs[ifaceName]
+	default:
+		return fmt.Errorf("unknown direction type")
+	}
+
+	if bpfList == nil {
+		log.Warn().Msgf("%s program list is empty", direction)
```

Comment:
Basically, this if statement will not be triggered because **AddProgramsOnInterface** will always push the root program before calling **AddAndStartBPF**  hence I am removing this statement.

Reply:
@Atul-source I think then we will need to handle the case of BpfChainingEnabled being disabled in AddProgramsOnInterface.  If chaining is disabled in the config, we should not insert a root program when the add / delete api is called.  It looks like delete handles this, but add will insert the root program still.